### PR TITLE
[WIP] Fix selecting filtered orders not working properly 

### LINKED ISSUE #2837: Selecting filtered orders not working properly
### Expected behavior

When there's a filter applied to the orders table and `select all` action performed, only the filtered orders should be selected.

### Actual Behavior

Even with a filter applied, `select all` selects all orders present in the table.

### Steps to Reproduce the Behavior

- Have a couple of orders in different states/stages
- Visit the orders dashboard (click on the orders icon in the sidebar)
- Apply any kind of filter to the orders e.g. search, paginate, filter by status
- Click `Select All` checkbox on order table header
- Observe that all the orders in the table are selected

### Versions
Node: 6.2.2
NPM: 3.10.9
Meteor Node: 4.8.4
Meteor NPM: 4.6.1
Reaction CLI: 0.11.0
Reaction: 1.5.0
Reaction branch: marketplace
Docker: 1.13.0

### PR DESCRIPTION
**Pull request by aanchirinah**
Resolves #2837

This PR fixes a bug on the orders page where when a filter is applied, `select all` selects all orders present in the table, including those not relevant to the filter.

#### Test Instructions
- Select the orders from the side nav bar.
- Select a filter and click the select all Checkbox, you will observe it displays the actual number of selected items.

### CODE REVIEW

File: imports/plugins/core/orders/client/containers/orderDashboardContainer.js [hunk 1]
```diff
@@ -155,9 +155,14 @@ class OrderDashboardContainer extends Component {
   }
 
   handleMenuClick = (event, value) => {
+  //  const query = OrderHelper.makeQuery(value);
```
Comment (zenweasel) [inline imports/plugins/core/orders/client/containers/orderDashboardContainer.js]: Don't comment out code. If we don't need it, remove it.

File: imports/plugins/core/orders/client/containers/orderDashboardContainer.js [hunk 2]
```diff
@@ -256,6 +261,26 @@ class OrderDashboardContainer extends Component {
         selectedItems: [],
         multipleSelect: false
       });
+    } else if (orders === []) {
+      // if filter is selected it resets all orders that has been checked by
```
Comment (zenweasel) [inline imports/plugins/core/orders/client/containers/orderDashboardContainer.js]: typo in comment

**Comment by zenweasel** [comment]
So something is still definitely wrong here. You can see from the screen capture that when I filter it is selecting orders when I selected none, and selecting more orders than are available in the filtered view. I _think_ what might have started that was that I selected All and then applied a filter, and then from then on nothing was working right.

This needs more testing.

order_select

**Comment by aanchirinah** [comment]
@zenweasel , I have fix the changes you requested for.

**Comment by zenweasel** [comment]
@aanchirinah I don't see any changes that would have fixed the bug I pointed out?

**Comment by aanchirinah** [comment]
@zenweasel , so the bug you experienced at first was from a commit I made earlier, I realised I hadn't pushed some commits until this morning. I resolved the issue you experienced at line 163 in `orderDashboardContainer.js`.

**Comment by kieha** [comment]
Selecting paginated orders doesn't work. I've filtered the orders to 5 per page, but on select still selects all 11 orders that I have.

screen shot 2017-09-18 at 8 54 21 pm

**Comment by zenweasel** [comment]
@aanchirinah What's the status on this?

**Comment by aanchirinah** [comment]
@zenweasel , collaborating with @joykare for a permanent fix

**Comment by joykare** [comment]
Its not only happening when filter is on, also happens when orders are paginated. It selects all possible orders as opposed to just the orders on the page.

Problem: https://github.com/reactioncommerce/reaction/blob/3d312a6e7938b94e725bcc66ead737f760279756/imports/plugins/core/orders/client/containers/orderDashboardContainer.js#L245
https://github.com/reactioncommerce/reaction/blob/marketplace/imports/plugins/core/orders/client/components/orderTable.js#L252

`selectAllOrders` fetches orders from the DB while all other order functions like pagination and filtering generate orders in the sortablTable.

https://github.com/reactioncommerce/reaction/blob/marketplace/imports/plugins/core/ui/client/components/table/sortableTable.js#L47

#2768 this issue should fix this. So I propose closing this issue in favor of #2768

**Comment by spencern** [comment]
@joykare @aanchirinah What's the status here? Need this fixed ASAP

**Comment by joykare** [comment]
@aanchirinah I'm not sure how far you got with this

@spencern picking up #2768 now, I have referenced the bugs it solves. This one is one of them. Will update its status as I work on it.

**Comment by spencern** [comment]
Sounds good. Thanks @joykare!

**Comment by spencern** [comment]
@joy - I've added the [WIP] prefix tag to this PR for my sanity. Please remove it once you're ready for a review.

**Comment by joykare** [comment]
Closing this PR in favor of #2972

**Pull request closed without merging**